gammapy/cube/make.py

```diff
@@ -33,196 +33,313 @@ class MapMakerObs:
         Reference image geometry in true energy, used for IRF maps. It can have a coarser
         spatial bins than the counts geom.
         If none, the same as geom is assumed
-    exclusion_mask : `~gammapy.maps.Map`
-        Exclusion mask (used by some background estimators)
     migra_axis : `~gammapy.maps.MapAxis`
         Migration axis for edisp map
     rad_axis : `~gammapy.maps.MapAxis`
-        Radial axis for psf map
+        Radial axis for psf map.
+    cutout : bool
+        Whether to cutout the observation.
     """
 
     def __init__(
         self,
-        observation,
         geom,
         offset_max,
         geom_true=None,
-        exclusion_mask=None,
         background_oversampling=None,
         migra_axis=None,
         rad_axis=None,
         cutout=True,
     ):
-
-        cutout_kwargs = {
-            "position": observation.pointing_radec,
-            "width": 2 * Angle(offset_max),
-            "mode": "trim",
-        }
-
-        if cutout:
-            geom = geom.cutout(**cutout_kwargs)
-            if geom_true is not None:
-                geom_true = geom_true.cutout(**cutout_kwargs)
-            if exclusion_mask is not None:
-                exclusion_mask = exclusion_mask.cutout(**cutout_kwargs)
-
-        self.observation = observation
+        self.cutout = cutout
         self.geom = geom
         self.geom_true = geom_true if geom_true else geom.to_binsz(BINSZ_IRF)
         self.offset_max = Angle(offset_max)
-        self.exclusion_mask = exclusion_mask
         self.background_oversampling = background_oversampling
-        self.maps = {}
         self.migra_axis = migra_axis if migra_axis else MIGRA_AXIS_DEFAULT
         self.rad_axis = rad_axis if rad_axis else RAD_AXIS_DEFAULT
 
-    def _fov_mask(self, coords):
-        pointing = self.observation.pointing_radec
-        offset = coords.skycoord.separation(pointing)
-        return offset >= self.offset_max
+    def _cutout_geom(self, geom, observation):
+        if self.cutout:
+            cutout_kwargs = {
+                "position": observation.pointing_radec,
+                "width": 2 * self.offset_max,
+                "mode": "trim",
```

Comment:
Shouldn't the cutout mode be an argument?
If we are to perform ring background later on, we cannot use `trim`.

Reply:
Sure, I can make it an argument. What was the reason again, that we cannot use partially contained observations for the ring background?

Reply:
yes, if the observation is partially contained you're likely to get an unexpected behavior.

Reply:
Is there a more specific explanation, that I could put in the docstring? I guess technically it still works, but it comes with increased systematics at the edge of the image?

Reply:
Yes, it also increases the probability of having empty OFF pixels

Reply:
I've added the `cutout_mode` option. I think general we should maybe re-discuss the default choice,  so that we compute the whole observation by default. I can make a separate issue for this....